source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -413,6 +413,15 @@ absl::optional<SystemTime> SslSocket::expirationPeerCertificate() const {
   return Utility::getExpirationTime(*cert);
 }
 
+std::string SslSocket::sessionId() const {
+  SSL_SESSION* session = SSL_get_session(ssl_.get());
```

Comment:
Check for null pointer? I don't claim any expertise in SSL APIs, but the man page implies it's possible for there to be no session.

Reply:
Good call -- done.